web3.js/src/transaction.js

```diff
@@ -268,44 +265,38 @@ export class Transaction {
       }
     });
 
-    // Move payer to the front and append other unknown signers as read-only
+    // Build ordered signed keys and add missing metas
+    const signedKeys: string[] = [];
     this.signatures.forEach((signature, signatureIndex) => {
```

Comment:
I think we need to actually sort `this.signatures` by `uniqueMetas` rather than the other way around.  There are runtime sanity checks (admittedly loose ATM), but I think it may still be possible to create a runtime incompatible ordering. That is mixed RO/RW signers.  Does that seem reasonable?

https://github.com/solana-labs/solana/blob/91d5b0b30c642bedf4d6a8d333b56f5e396cb453/sdk/src/message.rs#L182
https://github.com/solana-labs/solana/blob/91d5b0b30c642bedf4d6a8d333b56f5e396cb453/runtime/src/message_processor.rs#L523

Reply:
Ah dang, totally missed that distinction between RO/RW. Thanks!

Reply:
This may have been the root of a miscommunication in the last PR 🙃

My bad!